### PR TITLE
RUM-4829 Upgrade CI stack to Xcode 15

### PR DESCRIPTION
**Pull request by ncreated**
### What and why?

📦 This PR updates the repo so we can run Xcode 15 on CI.

It follows earlier https://github.com/DataDog/dd-sdk-ios/pull/1891 where we dropped iOS and tvOS 11.0 deployment targets (no longer available in Xcode 15).

### How?

3 main changes:
- **Aligned Swift version to 5.9 which comes with Xcode 15.0**. To make this configuration central, I removed all Xcode project overrides and defined it in `Base.xcconfig` (following the pattern from #1891). `Package.swift` and CP podspecs are updated accordingly.
- **Re-generated SR snapshot using iPhone 15 Simulator and iOS 17.5**. New snapshots can be seen in https://github.com/DataDog/dd-mobile-session-replay-snapshots/commit/5d8f4ffffabaa507dadc07bd89498bb962a64853 (internal). I did it a deeper look using diffusion diffs to make sure there is no visible regression.
- **Default CI stack will be changed to Xcode 15**. This will happen before merge.

Testing:
- I ran each stage (unit tests, SR snapshots, integration tests, smoke tests, tools) in isolation using temporary Xcode 15 stack. All worked fine ✅ ✅ ✅ ✅ ✅.
- I don't expect any impact on release automation because smoke tests for all dependency managers worked fine (including XCFrameworks build).

### Review checklist
- [ ] Feature or bugfix MUST have appropriate tests (unit, integration)
- [ ] Make sure each commit and the PR mention the Issue number or JIRA reference
- [ ] Add CHANGELOG entry for user facing changes

### Custom CI job configuration (optional)
- [ ] Run unit tests for Core, RUM, Trace, Logs, CR and WVT
- [ ] Run unit tests for Session Replay
- [ ] Run integration tests
- [ ] Run smoke tests
- [ ] Run tests for `tools/`

### CODE REVIEW

**Review by maciejburda — APPROVED** [review]

**Review by mariedm — APPROVED** [review]
🙌🚀

File: Package.swift [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// swift-tools-version: 5.7.1
```
Comment (ganeshnj) [inline Package.swift:1]: nice

File: IntegrationTests/xcconfigs/Base.xcconfig [hunk 2]
```diff
@@ -2,6 +2,9 @@
 IPHONEOS_DEPLOYMENT_TARGET=12.0
 TVOS_DEPLOYMENT_TARGET=12.0
 
+// Minimum supported Swift version
+SWIFT_VERSION=5.9
```
Comment (ganeshnj) [inline IntegrationTests/xcconfigs/Base.xcconfig:6]: nice

**Review by ganeshnj — APPROVED** [review]
What is the minimum version of the iOS simulator we are testing on in CI?

**Comment by ncreated** [comment]
> What is the minimum version of the iOS simulator we are testing on in CI?

@ganeshnj we use `iPhone 11,OS=latest` (see in [`Bitrise.yml`](https://github.com/DataDog/dd-sdk-ios/blob/develop/bitrise.yml)). Effectively it means iOS 17.5 for the current stack. We can look into distinguishing "high" and "low" simulators once we move to GitLab.

**Pull request merged**